### ISSUE TITLE
Misleading error on import under certain circumstances

### ISSUE THREAD

**Issue by impala2** [Kozea/cairocffi#95]
I ran into a situation that was rather difficult to debug. When the library is imported it tries to load the native cairo library, and suppresses any `OSError`s raised in the process. In the end it kept complaining that the native library was not present when it clearly was.

> dlopen() failed to load a library: cairo / cairo-2

The issue was that the server was running out of memory when trying to load the native library, but because this produces an `OSError`, it was silently suppressed.

    def dlopen(ffi, *names):
        """Try various names for the same library, for different platforms."""
        for name in names:
            for lib_name in [name, 'lib' + name]:
                try:
                    path = ctypes.util.find_library(lib_name)
                    if path:
                        lib = ffi.dlopen(path)
                        if lib:
                            return lib
                except OSError: # <-- this caused the issue
                    pass
        raise OSError("dlopen() failed to load a library: %s" % ' / '.join(names))

Is it possible to somehow make the exception catching more specific to avoid silencing unrelated problems? For example checking the error message, and if it's not related to library not being found, re-raise the exception?

Example solution:

    def dlopen(ffi, *names):
        """Try various names for the same library, for different platforms."""
        for name in names:
            for lib_name in [name, 'lib' + name]:
                try:
                    path = ctypes.util.find_library(lib_name)
                    if path:
                        lib = ffi.dlopen(path)
                        if lib:
                            return lib
                except OSError as ex:
                    if 'no such file' not in str(ex).lower():
                        raise
        raise OSError("dlopen() failed to load a library: %s" % ' / '.join(names))

Thanks.

**Comment by Tontyna** [comment]
AFAIK `ctypes.util.find_library` never raises an exception. And I'm asking myself what's the use of suppressing OSExceptions raised by `ffi.dlopen`? Such an exception indicates that the library **was found** but something ugly (maybe a missing dependency or lack of memory) prevented it from loading. 
Cannot think of a situation where suppressing that and continue with the next `lib_name` is a good solution. So why not let it raise and tell the truth?

```python
def dlopen(ffi, *names):
    """Try various names for the same library, for different platforms."""
    for name in names:
        for lib_name in [name, 'lib' + name]:
            path = ctypes.util.find_library(lib_name)
            if path:
                lib = ffi.dlopen(path)
                if lib:
                    return lib
    raise OSError("dlopen() failed to find a library: %s" % ' / '.join(names))
```

**Comment by liZe** [comment]
Closed by #135.
